### PR TITLE
feat(core): DSPX-896 add registered resources CRUD

### PR DESCRIPTION
**Pull request by ryanulit**
Adds CLI support for registered resources CRUD, including action attribute values as well

### CODE REVIEW

File: docs/man/policy/registered-resources/create.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+A registered resource `name` is normalized to lower case and may contain hyphens or dashes between other alphanumeric characters.
+
+For more information, see the `registered-resources` subcommand.
+
+## Examples
+
+Create a registered resource named 'my_resource' with value 'my_value':
+
+```shell
+otdfctl policy registered-resources create --name my_resource --v my_value
```
Comment (jakedoublev) [inline docs/man/policy/registered-resources/create.md:34]: The shorthand flags must have only one `-` here

```suggestion
otdfctl policy registered-resources create --name my_resource --value my_value
```

File: docs/man/policy/registered-resources/values/update.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+If PEPs rely on this value, a value update could break access.
+
+Make sure you know what you are doing.
+
+For more information about Registered Resource Values, see the manual for the `values` subcommand.
+
+## Example
+
+```shell
```
Comment (jakedoublev) [inline docs/man/policy/registered-resources/values/update.md:34]: Could you please add some examples and document the update behavior of fully replacing or removing the action attribute values on a registered resource value?

File: e2e/registered-resources.bats [hunk 3]
```diff
@@ -0,0 +1,388 @@
...
+    [ "$(echo "$output" | jq -r '.action_attribute_values[1].attribute_value.id')" = "$ATTR_VAL_2_ID" ]
+    [ "$(echo "$output" | jq -r '.action_attribute_values[1].attribute_value.fqn')" = "$ATTR_VAL_2_FQN" ]
+  created_id_with_action_attr_vals=$(echo "$output" | jq -r '.id')
+
+  # cleanup
+  run_otdfctl_reg_res_values delete --id $created_id_simple --force
+  run_otdfctl_reg_res_values delete --id $created_id_with_action_attr_vals --force
+}
+
+@test "Create a registered resource value - Bad" {
```
Comment (jakedoublev) [inline e2e/registered-resources.bats:234]: Can you please add test cases for your delimiter `;` that's good/bad so we can detect regressions if anything changes if anything around the delimiter changes going forward?

File: pkg/handlers/registeredResources.go [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+	}
+
+	return resp.GetValues(), resp.GetPagination(), nil
+}
+
+func (h Handler) UpdateRegisteredResourceValue(ctx context.Context, id, value string, actionAttributeValues []*registeredresources.ActionAttributeValue, metadata *common.MetadataMutable, behavior common.MetadataUpdateEnum) (*policy.RegisteredResourceValue, error) {
+	_, err := h.sdk.RegisteredResources.UpdateRegisteredResourceValue(ctx, &registeredresources.UpdateRegisteredResourceValueRequest{
+		Id:                     id,
+		Value:                  value,
+		ActionAttributeValues:  actionAttributeValues,
```
Comment (jakedoublev) [inline pkg/handlers/registeredResources.go:141]: Can you please add a test to ensure the pflags implementation doesn't end up passing an empty slice instead of a nil slice to this request and therefore clearing out the stored `actionAttributeValues`?

File: e2e/registered-resources.bats [hunk 5]
```diff
@@ -0,0 +1,388 @@
...
+    [ "$(echo "$output" | jq -r '.id')" != "" ]
+    [ "$(echo "$output" | jq -r '.value')" = "test_create_rr_val_with_action_attr_vals" ]
+    [ "$(echo "$output" | jq -r '.action_attribute_values[0].action.id')" = "$READ_ACTION_ID" ]
+    [ "$(echo "$output" | jq -r '.action_attribute_values[0].action.name')" = "$READ_ACTION_NAME" ]
+    [ "$(echo "$output" | jq -r '.action_attribute_values[0].attribute_value.id')" = "$ATTR_VAL_1_ID" ]
+    [ "$(echo "$output" | jq -r '.action_attribute_values[0].attribute_value.fqn')" = "$ATTR_VAL_1_FQN" ]
+    [ "$(echo "$output" | jq -r '.action_attribute_values[1].action.id')" = "$CUSTOM_ACTION_ID" ]
+    [ "$(echo "$output" | jq -r '.action_attribute_values[1].action.name')" = "$CUSTOM_ACTION_NAME" ]
+    [ "$(echo "$output" | jq -r '.action_attribute_values[1].attribute_value.id')" = "$ATTR_VAL_2_ID" ]
+    [ "$(echo "$output" | jq -r '.action_attribute_values[1].attribute_value.fqn')" = "$ATTR_VAL_2_FQN" ]
```
[the remark this answers is not in the document]
Comment (ryanulit) [reply e2e/registered-resources.bats]: `any` function works great, replaced with that

File: pkg/handlers/registeredResources.go (hunk 4 above)
Comment (ryanulit) [reply pkg/handlers/registeredResources.go:141]: Good catch here. Turns out there's a bug in my platform code that will delete the existing action attribute values regardless of whether any were passed in the request:

https://github.com/opentdf/platform/blob/main/service/policy/db/registered_resources.go#L433-L437

I'll add a piece to the update test here in bats, but comment it out until the bug is fixed in platform.

File: cmd/policy-registeredResources.go [hunk 6]
```diff
@@ -0,0 +1,610 @@
...
+// Registered Resource Values
+//
+
+func policyCreateRegisteredResourceValue(cmd *cobra.Command, args []string) {
+	c := cli.New(cmd, args)
+	h := NewHandler(c)
+	defer h.Close()
+
+	ctx := cmd.Context()
+	resourceID := c.Flags.GetRequiredID("resource-id")
```
[the remark this answers is not in the document]
Comment (ryanulit) [reply cmd/policy-registeredResources.go:205]: Which new commands use that? I copied this directly from attribute values.

[the remark this answers is not in the document]
Comment (jakedoublev) [reply cmd/policy-registeredResources.go:205]: Looks like [kas grants](https://github.com/opentdf/otdfctl/blob/c5524248130f480dd57cc234f125f746dfeb473b/cmd/kas-grants.go#L172-L180) and [subject mappings](https://github.com/opentdf/otdfctl/blob/c5524248130f480dd57cc234f125f746dfeb473b/cmd/policy-subjectMappings.go#L223-L227).

**Review by jakedoublev — DISMISSED** [review]

File: pkg/handlers/registeredResources.go (hunk 4 above)
Comment (ryanulit) [reply pkg/handlers/registeredResources.go:141]: Bug is fixed in the platform and the code added to the update test is now back to uncommented and being ran

**Review by jentfoo — APPROVED** [review]
LGTM

File: cmd/policy-registeredResources.go [hunk 7]
```diff
@@ -0,0 +1,633 @@
...
+
+	simpleRegResValues := cli.GetSimpleRegisteredResourceValues(resource.GetValues())
+
+	rows := [][]string{
+		{"Id", resource.GetId()},
+		{"Name", resource.GetName()},
+		{"Values", cli.CommaSeparated(simpleRegResValues)},
+	}
+
+	if mdRows := getMetadataRows(resource.GetMetadata()); mdRows != nil {
```
[the remark this answers is not in the document]
Comment (ryanulit) [reply cmd/policy-registeredResources.go:49]: This same code is used in other cmd logic, so will just leave it as-is for this PR. Appreciate the call out though.
